File: C_Training/FOP_DAY4/numOfDigs.c
/*Write a program to find the number of digits in a number using recursion.



Input Format

Input consists of a non-negative integer.



Output Format

Refer sample output for formatting specifications.



Sample Input 1

432



Sample Output 1

The number of digits in 432 is 3*/
#include <stdio.h>

int countDigits(int n) {
    if (n == 0)
        return 0;
    return 1 + countDigits(n / 10);
}

int main() {
    int num;

    scanf("%d", &num);

    printf("The number of digits in %d is %d",num, countDigits(num));

    return 0;
}

File: Java_Training/Recursion/numOfDigs.java
package Java_Training.Recursion;

import java.util.Scanner;

class Main {
    
    public static int countDigits(int n) {
        if (n == 0)
            return 0;
        return 1 + countDigits(n / 10);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int num = scanner.nextInt();

        System.out.println("The number of digits in " + num + " is " + countDigits(num));
        scanner.close();
    }
}
